tamr_client/datasets/record.py

```diff
@@ -0,0 +1,122 @@
+"""
+See https://docs.tamr.com/reference/record
```

Comment:
As part of this module-level docstring, we could include something like:

"The recommended approach for interacting with records is to use the `upsert` and `delete` functions for all use cases they can handle. For more advanced use cases, the underlying `_update` function can be used directly."

Note: I stole the phrasing from the standard library `subprocess` docs: https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module.

We can also copy this in the docstring for `_update` and replace the current wording to be consistent.

Reply:
I like that phrasing. I don't know if there are any changes I would make to the `_update` docstring.

Reply:
Can you give `upsert`, `delete` and `_update` ``the :func:`~tamr_client.dataset.upsert_records``` treatment?

Reply:
Done.